### DISCUSSION TITLE
Hi team I am trying to run Ginger automation in my local and i have VPN in place and i am not able to on Java app agent in local and its throwing below error if i am trying to test my javadriver agent connectionMethod - <ConnectToJavaDriver>b__36_0, Error - No connection could be made because the target machine actively refused it 127.0.0.1:8888

### DISCUSSION THREAD

**Discussion by aa92524** [Ginger-Automation/Ginger#2150] [General]
Hi team I am trying to run Ginger automation in my local and i have VPN in place and i am not able to on Java app agent in local and its throwing below error if i am trying to test my javadriver agent connectionMethod - <ConnectToJavaDriver>b__36_0, Error - No connection could be made because the target machine actively refused it 127.0.0.1:8888

**Comment by JinendraGhodke** [comment]
For Java agent, you need to launch the java application and attach the Ginger Java agent using below action.
The agent will start only after the attach agent is done.

https://ginger-automation.github.io/Ginger-Web-Help/assets/Ginger_By_Amdocs/Business_Flows/Actions/General_Action_Types/Launch_Java_Application_Action.htm?rhsearch=java%20drver&rhhlterm=java%20java%E2%80%99%20%E2%80%98java
